lesson21/index.js

```diff
@@ -110,5 +112,70 @@ const createTd = (usersData, keys) => {
   return fragment;
 }
 
+const createSortBtn = () => {
+  const sortBtn = createElementWithClassName("button", "sort-btn");
+  sortBtn.dataset.sortStatus = "default";
+  sortBtn.id = "js-sort-btn";
+  return sortBtn;
+}
+
+const setClickInSortBtn = () => {
+  const defaultRows = [...document.querySelector("tbody").querySelectorAll("tr")];
+  const sortBtn = document.getElementById("js-sort-btn");
+
+  sortBtn.addEventListener("click", (e) => {
+    const nextStatus = switchSortStatus(e.target.dataset.sortStatus);
+    e.target.dataset.sortStatus = nextStatus;
+
+    const sortedRows = getSortedRows(nextStatus, defaultRows,e.target);
+
+    const tbody = document.querySelector("tbody");
+    sortedRows.forEach((row) => {
+      tbody.appendChild(row);
+    })
+  });
+}
+
+const switchSortStatus = (status) => {
+  switch (status) {
+    case "default":
+      return "asc";
+
+    case "desc":
+      return "default";
+
+    case "asc":
+      return "desc";
+
+    default:
+      return "default";
+  }
+}
+
+const findClickedCellIndex = (target) => {
+  const th = [...document.querySelector("thead").querySelectorAll("th")];
+  return th.indexOf(target.parentElement);
+}
+
+const getSortedRows = (status, defaultLows,target) => {
+  if (status === "default") {
+    return defaultLows;
+  }
+  return sortByClickedCell(status, defaultLows,findClickedCellIndex(target));
+}
+
+const sortByClickedCell = (status, defaultLows, index) => {
+  if (status === "asc") {
+    return [...defaultLows].sort(
+      (a, b) => a.children[index].textContent - b.children[index].textContent
+    );
+  }
+  if (status === "desc") {
+    return [...defaultLows].sort(
+      (a, b) => b.children[index].textContent - a.children[index].textContent
+    );
+  }
+};
```

Comment:
You made it one function in descending and descending order! It's very readable.

Reply:
Thank you for everything.
I am always encouraged by your comments ☺️